netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -195,8 +193,13 @@ private void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
           checkNotNull(transportListener.methodDetermined(method, metadata), "statsTraceCtx");
       NettyServerStream.TransportState state = new NettyServerStream.TransportState(
           this, http2Stream, maxMessageSize, statsTraceCtx);
-      NettyServerStream stream = new NettyServerStream(ctx.channel(), state, attributes,
+      Attributes attributesWithAuthority = Attributes
+              .newBuilder()
+              .setAll(attributes).set(ServerImpl.ATTR_AUTHORITY, headers.authority().toString())
```

Comment:
Is there a cleaner way to get `authority` up to the `callbackRegistry` that doesn't involve passing `authority` along the call path?